activesupport/lib/active_support/notifications/fanout.rb

```diff
@@ -113,11 +116,33 @@ def self.wrap_all(pattern, subscriber)
           end
         end
 
+        class Matcher #:nodoc:
+          attr_reader :pattern, :blacklist
+
+          def self.wrap(pattern)
+            return pattern if String === pattern
+            new(pattern)
+          end
+
+          def initialize(pattern)
+            @pattern = pattern
+            @blacklist = Set.new
+          end
+
+          def blacklist!(name)
+            blacklist << -name if pattern === name
+          end
+
+          def ===(name)
+            pattern === name && blacklist.none? { |p| p === name }
```

Comment:
Can we use `Set#include?` instead? Since we know name and everything in the set to be a string